components/datetime/src/pattern/mod.rs

```diff
@@ -70,7 +71,13 @@ fn get_time_granularity(item: &PatternItem) -> Option<TimeGranularity> {
         PatternItem::Field(field) => match field.symbol {
             fields::FieldSymbol::Hour(_) => Some(TimeGranularity::Hours),
             fields::FieldSymbol::Minute => Some(TimeGranularity::Minutes),
-            fields::FieldSymbol::Second(_) => Some(TimeGranularity::Seconds),
+            fields::FieldSymbol::Second(fields::Second::Second) => Some(TimeGranularity::Seconds),
+            fields::FieldSymbol::Second(fields::Second::Millisecond) => {
+                Some(TimeGranularity::Seconds)
+            }
```

Comment:
nitpick: Don't match on the `A` field.  We should just ignore it.  The only ones we care about are `s` (second) and `S` (fractional seconds).